test/test_public_bindings.py

```diff
@@ -370,6 +370,7 @@ def test_modules_can_be_imported(self):
             "torch._inductor.codegen.cuda.cuda_template",
             "torch._inductor.codegen.cuda.gemm_template",
             "torch._inductor.runtime.triton_helpers",
+            "torch._inductor.runtime.halide_helpers",
```

Comment:
The fact that you have it here means that your code to try and handle the case where halide is not supported in halide_helpers is not working as expected.

You might want to fix that file to properly work when halide is not available?

Reply:
Yes you're right. The reason why I didn't handle the import exception in `halide_helpers` in the first place was that `triton_helpers` does not handle it, where I believe that's the reason that `triton_helpers` import test is skipped here. @jansel suggested that we should handle the exception for halide: https://github.com/pytorch/pytorch/pull/130211#discussion_r1669468484.

So my question is: why is there a discrepancy here where triton_helpers doesn't handle the exception and adds a skip here while we want halide_helpers to handle it?

Reply:
We should make both of those files import cleanly without halide/triton and remove them here.

Reply:
Oh I somehow got to know why we import triton directly without putting it in a try catch: triton relies on `tl.constexpr` hint to annotate constexpr. If we put the import in try catch and set `None` when triton package is not found, then mypy will complain (and python interpreter errors out if we run the file).